### DISCUSSION TITLE
Program type - Interval

### DISCUSSION THREAD

**Discussion by rmperreault** [vinteo/hass-opensprinkler#168] [Q&A] [answered]
I am using the program type interval - when the program runs on automatic the "starting in" gets reset to the Interval days i.e. if the interval days is 7 then when the program runs on schedule the starting in gets reset to 7 at midnight.  Is there a way to do the same from Home Assistant meaning if i call the service to run a program it would reset the same way?  
I am trying to use remote soil sensors that would trigger a program under a set level but still have the interval backup plan in the event the sensors go dead on me.
Thanks for any help.

**Comment by vinteo** [comment, marked as answer]
Maybe test if disabling and then enabling the program resets the interval?
There is a switch for that.

Otherwise you might have to move the scheduling to home assistant some how
rather than use the built in program on the device

**Reply by rmperreault** [reply]
i did try this already.  
i did find an older document from firmware 1.8 but being new at this i am not sure how to implement it via homeassistant.  Based on the example days1 = 1 would be the interval while days0=128 or 129 would be the number the starting point .  Any suggestions on how to setup a command in homeassistant?

6. Change (add/modify) programs (password required) [Keyword /cp]:
Command: /cp?pw=xxx&pid=x&v=[x,x,x,x,x,x,x]
Parameters:
● pw: password
● pid: program index (starting from 0). For example, 0 means the first program, and so on.
Must be less than the total number of programs. A value of -1 will add a new program.
● v: program values, consisting of 7 integers:
○ en: program enable bit (binary value, e.g. 0 means the program is diabled)
○ days0: days selection first byte (see explanation below)
○ days1: days selection second byte (see explanation below)
○ starttime: program start time (in minutes relative to 00:00. e.g. 360 means 6am)
○ endtime: program end time (in minutes relative to 00:00)
○ duration: duration of each station (in seconds)
○ interval: program interval time (in minutes. e.g. 240 means every 4 hours)
Days selection bytes:
days0 and days1 in the program values are formatted as follows:
● if (days0.bit7 == 0 || days1 == 0 || days1 == 1): this is a standard weekly program:
○ days0.bits[0..6] store the binary selection bit from Monday to Sunday (bit 0
corresponds to Monday)
○ if (days1 == 1): even day restriction
○ if (days1 == 2): odd day restriction
● if (days0.bit7 == 1 && days1 > 1): this is an interval program:
○ days1 stores the interval (>1. e.g. 2 means every 2 days, 3 means every 3 days
and so on)
○ days0.bits[0..6] store the starting day (e.g. 0 means the program starts today, 1
means the program starts tomorrow, and so on)
● Examples:
○ days0 = 127, days1 = 0: weekly schedule on every day
○ days0 = 255, days1 = 1: weekly schedule on every day, with odd day restriction
○ days0 = 128, days1 = 5: interval schedule, every 5 days, starting from today
○ days0 = 129, days1 = 4: interval schedule, every 4 days, starting from tomorrow

**Reply by vinteo** [reply]
Probably could be done using this:
https://www.home-assistant.io/integrations/rest_command/

**Reply by rmperreault** [reply]
thanks for your help, I will try this weekend.
